### PR TITLE
Nintendo Switch support

### PR DESCRIPTION
**Pull request by fgsfdsfgs**
This adds basic support for the Nintendo Switch again (target name `nswitch`). Based on the [old branch](https://github.com/FWGS/xash3d-fwgs/tree/switch_new).
CI scripts also included.

**Building instructions:**
1. Get `devkitA64` and `dkp-pacman`.
2. Install dependencies:
   ```
   dkp-pacman -S switch-dev dkp-toolchain-vars switch-sdl2
   git clone https://github.com/fgsfdsfgs/libsolder.git
   make -C libsolder install
   ```
3. Use `waf`:
   ```
   ./waf configure -T release --nswitch
   ./waf build
   ```

**NOTE**: The CI build script currently pulls HLSDK from [my own fork](https://github.com/fgsfdsfgs/hlsdk-xash3d). When the Switch-related changes are merged into `FWGS/hlsdk-portable`, you should change this in `deps_nswitch.sh`.

### CODE REVIEW

**Comment by SNMetamorph** [comment]
Good work

File: engine/platform/nswitch/sys_nswitch.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+#include <sys/stat.h>
+#include <switch.h>
+#include <solder.h>
+#include <SDL.h>
+
+static int nxlink_sock = -1;
+
+/* HACKHACK: force-export stuff required by the dynamic libs */
+
+// libunwind stuff for C++ exceptions, required by filesystem_stdio
```
Comment (a1batross) [inline engine/platform/nswitch/sys_nswitch.c:31]: C++ exceptions in filesystem_stdio weren't supposed to be enabled... 

But anyway, let's keep _Unwind symbols anyway for mod compatibility.

Comment (fgsfdsfgs) [reply engine/platform/nswitch/sys_nswitch.c]: Turns out the build system was linking libstdc++ to everything by mistake. Fixed in 69607d78.

File: common/port.h [hunk 2]
```diff
@@ -39,10 +39,15 @@ GNU General Public License for more details.
 
 	#if XASH_POSIX
 		#include <unistd.h>
-		#include <dlfcn.h>
-		#define HAVE_DUP
-		#define O_BINARY    0
-		#define O_TEXT      0
+		#ifdef XASH_NSWITCH
```
[the remark this answers is not in the document]
Comment (fgsfdsfgs) [reply common/port.h]: Fixed in 4e87eb0.

File: common/defaults.h [hunk 3]
```diff
@@ -177,4 +181,9 @@ Default build-depended cvar and constant values
 	#define DEFAULT_FULLSCREEN 1
 #endif // DEFAULT_FULLSCREEN
 
+#if XASH_NSWITCH
```
[the remark this answers is not in the document]
Comment (fgsfdsfgs) [reply common/defaults.h:184]: That would also always enable the `developer` cvar though, not sure if you'd want that. This is equivalent to passing `-console` on the command line, setting `DEFAULT_DEV` would be equivalent to passing `-dev`.

[the remark this answers is not in the document]
Comment (a1batross) [reply common/defaults.h:184]: Ah, yes, you're right.

**Comment by a1batross** [comment]
I tested the autobuild and it worked on my side. There are few inconsistensies like the Nintendo button mapping, touch enabled by default, can't get back to menu with just gamepad but these probably can be solved later.

I'm merging this.

**Pull request merged**